### PR TITLE
refactor: visitor journey list

### PR DESCRIPTION
**Pull request by csiyang**
# Description

### <samp>🤖 Generated by Copilot at cfce69d</samp>

This pull request adds the JourneyList component and its subcomponents, which display the timeline of events for a visitor's journey. It also adds the unit tests and storybook files for each component, and the utility functions to transform and sort the events data. The components use MUI, i18next, and GraphQL to match the design and functionality requirements. The main files affected are `JourneyList.tsx`, `EventsCard.tsx`, `TimelineEvent.tsx`, and `CompactEvent.tsx`.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/32387047/todos/6067267402)

# How should this PR be QA Tested?
Page path: `/reports/visitors/[visitorId]`
This PR just deals with UX of the `visitor journey card` in `journeys-admin`. Specific events UI will be updated in a future PR

- [ ] Show collapsed timeline (events that should show up when collapsed are: `ChatOpen`, `TextResponse`, `PollResponse`, and `JourneyView`
- [ ] Show expanded timeline - can expand through clicking `x more events` button. (events that should show up when expanded are the same from the collapsed list plus: `ButtonClick`, `VideoComplete`, `VideoStart`, and `SignUp`
- [ ] Should not let you collapse the timeline after expanding (unless the page is refreshed)
- [ ] Journey Started event has the time when the event was recorded
- [ ] Events in the timeline should display the duration spent on the event in a `m/ss` or `mm/ss` structure. Unless the duration is less than 1 second in which case, it should show `< 0:01`
- [ ] The last event of the timeline should have a blank duration
- [ ] Card header should have `total duration`, `journey title`, and `date first visited`

# Walkthrough

### <samp>🤖 Generated by Copilot at cfce69d</samp>

*  Implement the JourneyList component ([link](https://github.com/JesusFilm/core/pull/1518/files?diff=unified&w=0#diff-7fb67e3a200410e46c2078b8844e50aa0e471d3ba199cb1bde0c1754c52c3529R1-R58)) that fetches and renders the visitor events using the EventsCard component ([link](https://github.com/JesusFilm/core/pull/1518/files?diff=unified&w=0#diff-944413ed4953c52240777a109c2e2d34ff5cb592c113451495a749e72fb5201aR1-R94))

### CODE REVIEW

File: apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/utils/messagePlatformToLabel/messagePlatformToLabel.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { TFunction } from 'react-i18next'
+import { MessagePlatform } from '../../../../../../__generated__/globalTypes'
+
+export function messagePlatformToLabel(
```
Comment (csiyang) [inline apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/utils/messagePlatformToLabel/messagePlatformToLabel.ts:4]: this is just a copy from the existing ` messagePlatformToLabel`

File: apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/utils/transformToJourney/transformToJourney.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+export interface JourneyWithEvents {
+  id: string
+  title?: string | null
+  subtitle?: string
+  createdAt?: string
+  events: Event[]
+}
+
+export function transformToJourney(events: Event[] = []): JourneyWithEvents[] {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/utils/transformToJourney/transformToJourney.ts:12]: mostly unchanged from the original transformToJourney. Only added journeyViewEvent to be added in the list of events

File: apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/utils/videoBlockSourceToLabel/videoBlockSourceToLabel.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { TFunction } from 'react-i18next'
+import { VideoBlockSource } from '../../../../../../__generated__/globalTypes'
+
+export function videoBlockSourceToLabel(
```
Comment (csiyang) [inline apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/utils/videoBlockSourceToLabel/videoBlockSourceToLabel.ts:4]: unchanged from original

File: apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/utils/transformEvents/transformEvents.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  event: Event
+  duration?: string
+}
+
+interface TransformedEvents {
+  timelineItems: Array<TimelineItem | TimelineItem[]>
+  totalDuration: string
+}
+
+export function transformEvents(events: Event[]): TransformedEvents {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/utils/transformEvents/transformEvents.ts:16]: Calculates duration between events and nests certain events to easily show collapsed version of timeline

File: apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/utils/transformEvents/transformEvents.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  const durationInMilliseconds = differenceInMilliseconds(
+    parseISO(end),
+    parseISO(start)
+  )
+
+  const minutes = Math.floor((durationInMilliseconds % 3600000) / 60000)
+  const seconds = Math.floor(((durationInMilliseconds % 360000) % 60000) / 1000)
+
+  let result: string
+  if (minutes === 0 && seconds === 0) {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/utils/transformEvents/transformEvents.ts:89]: some events have a duration in the milliseconds. so we just generalise those to under 1 second

File: apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/utils/transformEvents/transformEvents.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+  GetVisitorEvents_visitor_events_ChatOpenEvent as ChatOpenEvent
+} from '../../../../../../__generated__/GetVisitorEvents'
+import { getDuration } from './transformEvents'
+import { TimelineItem, transformEvents } from '.'
+
+const journeyViewEvent: JourneyViewEvent = {
+  __typename: 'JourneyViewEvent',
+  id: 'journeyView1.id',
+  journeyId: 'journey1.id',
+  createdAt: '2022-11-02T03:20:26.368Z',
```
Comment (csiyang) [inline apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/utils/transformEvents/transformEvents.spec.tsx:20]: each createdAt value is intentionally set for duration calculation testing

File: apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/VisitorJourneysList.tsx [hunk 7]
```diff
@@ -46,31 +39,20 @@ export const GET_VISITOR_EVENTS = gql`
   }
 `
 
-export function VisitorJourneyList({ id, limit }: Props): ReactElement {
-  const {
-    state: { journey },
-    dispatch
-  } = useVisitorInfo()
+interface Props {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/VisitorJourneysList.tsx:42]: replaces the old journeyVisitorList

File: apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/EventsCard/TimelineEvent/TimelineEvent.tsx [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+} from '../../utils'
+import { GenericEvent } from '../GenericEvent'
+
+interface Props {
+  timelineItem: TimelineItem
+}
+
+type GenericEventProps = ComponentProps<typeof GenericEvent>
+
+export function TimelineEvent({ timelineItem }: Props): ReactElement {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/EventsCard/TimelineEvent/TimelineEvent.tsx:24]: Mostly unchanged from the original `VisitorJourneyTimelineItem`, will update the design of specific events in a future PR

File: apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/EventsCard/TimelineEvent/TimelineEvent.tsx [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+      icon = <AnnouncementIcon />
+      label = t('Chat Opened')
+      value = t('Chat started on {{messagePlatform}}', {
+        messagePlatform:
+          event.messagePlatform != null
+            ? messagePlatformToLabel(event.messagePlatform, t)
+            : t('Message Platform')
+      })
+      break
+    case 'JourneyViewEvent':
```
Comment (csiyang) [inline apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/EventsCard/TimelineEvent/TimelineEvent.tsx:90]: Added a case for journeyViewEvent

**Review by csiyang — COMMENTED** [review]
Created a new component based of the old one since there was going to be so many changes. A lot of file changes are just from moving files into the new component.

**Review by mikeallisonJS — APPROVED** [review]

File: apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/EventsCard/CompactEvent/CompactEvent.tsx [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+interface Props {
+  value: string
+  handleClick: () => void
+}
+
+export function CompactEvent({ value, handleClick }: Props): ReactElement {
+  return (
+    <TimelineItem
+      sx={{
+        minHeight: '40px',
```
[the remark this answers is not in the document]
Comment (csiyang) [reply apps/journeys-admin/src/components/VisitorInfo/VisitorJourneysList/EventsCard/CompactEvent/CompactEvent.tsx:18]: No, can't remember the specifics, but collapse needs this to display properly

**Pull request merged**